application/models/Survey.php

```diff
@@ -671,8 +671,10 @@ public function getAllLanguages()
      */
     public function getTokenAttributes()
     {
-        $attdescriptiondata = decodeTokenAttributes($this->attributedescriptions);
-
+        $attdescriptiondata = decodeTokenAttributes($this->attributedescriptions ?? '');
+        if (!is_array(reset($attdescriptiondata))) {
```

Comment:
Why not check it in https://github.com/LimeSurvey/LimeSurvey/blob/a089c468838c1ff90603b78c44a5296817c5e16b/application/helpers/common_helper.php#L4764 directly ?

Check if the array return is valid ? If not return an empty array ?

Reply:
Ok, 

No and more : maybe must move https://github.com/LimeSurvey/LimeSurvey/blob/a089c468838c1ff90603b78c44a5296817c5e16b/application/helpers/common_helper.php#L4764 in this function or create a private function here. 

And deprecate common_helper function